sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAsyncApiTests.java

```diff
@@ -1515,15 +1495,16 @@ public void listRangesLease() throws IOException {
 
     @Test
     public void listRangesLeaseFail() throws IOException {
-        primaryFileAsyncClient.createWithResponse(1024, null, null, null, null).block();
         String fileName = generatePathName();
         String uploadFile = FileShareTestHelper.createRandomFileWithLength(1024, testFolder, fileName);
-        primaryFileAsyncClient.uploadFromFile(uploadFile).block();
-        createLeaseClient(primaryFileAsyncClient).acquireLease().block();
 
-        StepVerifier
-            .create(primaryFileAsyncClient.listRanges(null,
-                new ShareRequestConditions().setLeaseId(testResourceNamer.randomUuid())))
+        StepVerifier.create(primaryFileAsyncClient.createWithResponse(1024, null, null, null, null)
+            .then()
+            .then(primaryFileAsyncClient.uploadFromFile(uploadFile).then())
+            .then(createLeaseClient(primaryFileAsyncClient).acquireLease())
+            .then(primaryFileAsyncClient
+                .listRanges(null, new ShareRequestConditions().setLeaseId(testResourceNamer.randomUuid()))
+                .then()))
```

Comment:
There are several redundant statements here - you can remove the `then()`'s and change the last `.then(` to `.thenMany(`

```
StepVerifier.create(primaryFileAsyncClient.createWithResponse(1024, null, null, null, null)
            .then(primaryFileAsyncClient.uploadFromFile(uploadFile))
            .then(createLeaseClient(primaryFileAsyncClient).acquireLease())
            .thenMany(primaryFileAsyncClient
                .listRanges(null, new ShareRequestConditions().setLeaseId(testResourceNamer.randomUuid()))))
            .verifyError(ShareStorageException.class);
```